### ISSUE TITLE
snapperd crashes on Arch Linux

### ISSUE THREAD

**Issue by PotcFdk** [openSUSE/snapper#394]
snapperd crashes every few days on Arch Linux (package snapper 0.5.4-1)

the last 5 crashes:
```Feb 07 03:41:48 desktop systemd-coredump[22169]: Process 22037 (snapperd) of user 0 dumped 
Feb 07 13:25:36 desktop systemd-coredump[1305]: Process 1085 (snapperd) of user 0 dumped core.
Feb 07 14:37:53 desktop systemd-coredump[4566]: Process 3737 (snapperd) of user 0 dumped core.
Feb 08 01:03:10 desktop systemd-coredump[31146]: Process 30189 (snapperd) of user 0 dumped core.
Feb 09 12:19:37 desktop systemd-coredump[2489]: Process 1396 (snapperd) of user 0 dumped core.
Feb 11 16:14:35 desktop systemd-coredump[21173]: Process 20813 (snapperd) of user 0 dumped core.
```

The full journalctl output of the last crash:

```
Feb 11 16:14:34 desktop org.opensuse.Snapper[583]: terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
Feb 11 16:14:34 desktop org.opensuse.Snapper[583]:   what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
Feb 11 16:14:34 desktop systemd[1]: Created slice system-systemd\x2dcoredump.slice.
Feb 11 16:14:34 desktop systemd[1]: Started Process Core Dump (PID 21172/UID 0).
Feb 11 16:14:35 desktop systemd-coredump[21173]: Process 20813 (snapperd) of user 0 dumped core.
                                                 
                                                 Stack trace of thread 20813:
                                                 #0  0x00007f7a4f92c860 raise (libc.so.6)
                                                 #1  0x00007f7a4f92dec9 abort (libc.so.6)
                                                 #2  0x00007f7a4ff54d57 _ZN9__gnu_cxx27__verbose_terminate_handlerEv (libstdc++.so.6)
                                                 #3  0x00007f7a4ff528c6 _ZN10__cxxabiv111__terminateEPFvvE (libstdc++.so.6)
                                                 #4  0x00007f7a4ff5179a __cxa_call_terminate (libstdc++.so.6)
                                                 #5  0x00007f7a4ff52189 __gxx_personality_v0 (libstdc++.so.6)
                                                 #6  0x00007f7a4fcbeca3 _Unwind_RaiseException_Phase2 (libgcc_s.so.1)
                                                 #7  0x00007f7a4fcbf4ce _Unwind_Resume (libgcc_s.so.1)
                                                 #8  0x00007f7a50948fbc _ZN7snapper14logStreamCloseENS_8LogLevelEPKcjS2_PNSt7__cxx1119basic_ostringstreamIcSt11char_traitsIcESaIcEEE (libsnapper.so.4)
                                                 #9  0x00007f7a509174bf _ZN7snapper7SnapperD1Ev (libsnapper.so.4)
                                                 #10 0x000056133ef1685c n/a (snapperd)
                                                 #11 0x000056133ef1699c n/a (snapperd)
                                                 #12 0x00007f7a4f92f448 __run_exit_handlers (libc.so.6)
                                                 #13 0x00007f7a4f92f49a exit (libc.so.6)
                                                 #14 0x00007f7a4f918f51 __libc_start_main (libc.so.6)
                                                 #15 0x000056133eefa30a n/a (snapperd)```

**Comment by alpinista82** [comment]
Yes, I can confirm this bug. Is there any solution to this yet? Have this happening frequently (several times per week) on two armv7h machines. Haven't seen this error message on my x86 machines yet.

**Comment by alpinista82** [comment]
Just reviewed the logs on my x86-64 machines and noticed several of these annoying errors there as well...so this bug is obviously architecture independent. Has anybody got a clue what exactly is causing it or even has a workaround or will I have to find out for myself?

**Comment by NicoHood** [comment]
Can you please post the exact commands you are using to extract the logs with the errors? Then I can also review my system.

Which kernel are you using?

**Comment by alpinista82** [comment]
Just use the filter on your journal: journalctl -t org.opensuse.Snapper. Right now I am using Kernel 4.17.2-1-ARCH, but the problem seems to be unrelated to kernel. Must have something to with libboost, but I have never used this library so my knowledge is limited. The corresponding error message looks like this: "org.opensuse.Snapper[$PID] : terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error" and "org.opensuse.Snapper[$PID] :  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument"

**Comment by NicoHood** [comment]
I only get tons of those lines:
```
Nov 13 18:49:43 talloniv org.opensuse.Snapper[493]: :1: parser error : Document is empty
Nov 13 18:49:43 talloniv org.opensuse.Snapper[493]: ^
```

**Comment by alpinista82** [comment]
The bug seems to be fixed by commit d610a3a. Can be closed now. Thanks, keep up the great work!

**Comment by aschnell** [comment]
Yes, the backtrace look like in a bug I fixed this week. Thanks for the feedback.
